File: UserManager.js
const User = require('./User');

class UserManager {
    constructor() {
        this.users = [];
        if (this.users.length === 0) {
            this.createUser("bryan", "lakara");
        }
    }

    createUser(username, password) {
        this.users.push(new User(username, password));
        console.log("Utilisateur créé avec succès !");
    }

    authenticate(username, password) {
        for (let user of this.users) {
            if (user.username === username && user.password === password) {
                return true;
            }
        }
        return false;
    }
}

module.exports = UserManager;

File: WalletManagmentJava/UserManager.java
import java.util.ArrayList;
import java.util.List;

public class UserManager {
    private List<User> users;

    public UserManager() {
        this.users = new ArrayList<>();
        // Ajoutez un utilisateur par défaut s'il n'y en a pas
        if (users.isEmpty()) {
            createUser("bryan", "lakara");
        }
    }

    public void createUser(String username, String password) {
        users.add(new User(username, password));
        System.out.println("Utilisateur créé avec succès !");
    }

    public boolean authenticate(String username, String password) {
        for (User user : users) {
            if (user.getUsername().equals(username) && user.getPassword().equals(password)) {
                return true;
            }
        }
        return false;
    }
}